rimage/transform/parallel_projection.go

```diff
@@ -95,5 +97,169 @@ func (pp *ParallelProjection) PointCloudToRGBD(cloud pointcloud.PointCloud) (*ri
+		minX = math.Min(minX, robotMarker.Point().X)
+		maxZ = math.Max(maxZ, robotMarker.Point().Z)
+		minZ = math.Min(minZ, robotMarker.Point().Z)
+	}
+
+	// Calculate the scale factors
+	widthScaleFactor := float64(imageWidth) / (maxX - minX)
+	heightScaleFactor := float64(imageHeight) / (maxZ - minZ)
+
+	// Add points in the pointcloud to a new image
+	color := rimage.NewImage(imageWidth, imageHeight)
+	cloud.Iterate(0, 0, func(pt r3.Vector, data pointcloud.Data) bool {
+		j := (pt.X - meta.MinX) * widthScaleFactor
+		i := (pt.Z - meta.MinZ) * heightScaleFactor
+		x, y := int(math.Round(i)), int(math.Round(j))
+
+		// Adds a point to an image using the value to define the color. If no value is available,
+		// the default color of white is used.
+		if x >= 0 && x < imageWidth && y >= 0 && y < imageHeight && data != nil {
+			var c rimage.Color
+			if data.HasValue() {
+				c = getProbabilityColorFromValue(data.Value())
+			} else {
+				c = rimage.NewColor(255, 255, 255)
+			}
+			addPointToImage(color, image.Point{x, y}, c, voxelSize)
+		}
+		return true
+	})
+
+	// Add a red robot marker to the image
+	// TODO: Represent the orientation of the robot in its marker
+	if ppRM.robotPose != nil {
+		robotMarkerPoint := image.Point{
+			X: int(math.Round((robotMarker.Point().X - meta.MinX) * widthScaleFactor)),
+			Y: int(math.Round((robotMarker.Point().Z - meta.MinZ) * heightScaleFactor)),
+		}
+		robotMarkerColor := rimage.NewColor(255, 0, 0)
+		addPointToImage(color, robotMarkerPoint, robotMarkerColor, voxelSize*robotMarkerRatio)
+	}
+	return color, nil, nil
+}
+
+// RGBDToPointCloud calls its equivalent in ParallelProjection.
+func (ppRM *ParallelProjectionOntoXZWithRobotMarker) RGBDToPointCloud(
+	img *rimage.Image,
+	dm *rimage.DepthMap,
+	crop ...image.Rectangle,
+) (pointcloud.PointCloud, error) {
+	pp := ParallelProjection{}
+	return pp.RGBDToPointCloud(img, dm, crop...)
+}
+
+// ImagePointTo3DPoint calls its equivalent in ParallelProjection.
+func (ppRM *ParallelProjectionOntoXZWithRobotMarker) ImagePointTo3DPoint(pt image.Point, d rimage.Depth) (r3.Vector, error) {
+	pp := ParallelProjection{}
+	return pp.ImagePointTo3DPoint(pt, d)
+}
+
+// getProbabilityColorFromValue returns an RGB color value based on the the probability value and defined hit and miss
+// thresholds
+// TODO (RSDK-1074): once probability values are available this function should be changed to produced desired images.
+func getProbabilityColorFromValue(v int) rimage.Color {
+	var r, g, b uint8
+
+	prob := float64(v) / 100.
+
+	switch {
+	case prob < missThreshold && prob >= 0:
+		b = uint8(255 * (prob / missThreshold))
+	case prob > hitThreshold && prob <= 0:
+		g = uint8(255 * ((prob - hitThreshold) / (1 - hitThreshold)))
+	default:
+		r = 255
+		b = 255
+		g = 255
+	}
+
+	return rimage.NewColor(r, g, b)
+}
+
+// Calculates the mean and standard deviation of the provided data.
+func calculateMeanAndStandardDeviation(data []float64) (float64, float64, error) {
```

Comment:
[opt] I don't think there's that much advantage to having this helper vs just calling the underlying functions.

Reply:
I thought it would avoid repeats in code but I can remove the helper